### ISSUE TITLE
Can only download malformed zip files

### ISSUE THREAD

**Issue by fin444** [ViRb3/serve-zip#66]
```
$ mkdir /home/fin444/testdir
$ echo "some contents" > /home/fin444/testdir/testfile
$ wget https://github.com/ViRb3/serve-zip/releases/download/v1.0.3/serve-zip_1.0.3_linux_amd64
$ chmod +x serve-zip_1.0.3_linux_amd64
$ ./serve-zip_1.0.3_linux_amd64 "/home/fin444/" --port=35211 --host="localhost" &
$ wget localhost:35211/testdir/?zip=1
```

When running the above commands (or doing it normally through the browser w/ nginx, but this is more reproducible), I end up with a malformed zip file. It does have "some contents" in the raw data (see below), but when I try to open it in an archive manager things fall apart. Ark reports that it contains two zero-size files named `testdir`, and `unzip` gives the following output:

```
Archive:  index.html?zip=1
 extracting: testdir                 
checkdir error:  testdir exists but is not directory
                 unable to process testdir/testfile.

1 archive had fatal errors.
```

Here is a hex dump:

```
$ xxd index.html\?zip\=1
00000000: 504b 0304 1400 0800 0000 6ba2 4857 0000  PK........k.HW..
00000010: 0000 0000 0000 0000 0000 0700 0900 7465  ..............te
00000020: 7374 6469 7255 5405 0001 ca0e 2365 504b  stdirUT.....#ePK
00000030: 0708 0000 0000 0000 0000 0000 0000 504b  ..............PK
00000040: 0304 1400 0800 0000 6ba2 4857 0000 0000  ........k.HW....
00000050: 0000 0000 0000 0000 1000 0900 7465 7374  ............test
00000060: 6469 722f 7465 7374 6669 6c65 5554 0500  dir/testfileUT..
00000070: 01ca 0e23 6573 6f6d 6520 636f 6e74 656e  ...#esome conten
00000080: 7473 0a50 4b07 08cd b81e ac0e 0000 000e  ts.PK...........
00000090: 0000 0050 4b01 0214 0314 0008 0000 006b  ...PK..........k
000000a0: a248 5700 0000 0000 0000 0000 0000 0007  .HW.............
000000b0: 0009 0000 0000 0000 0010 00e8 4100 0000  ............A...
000000c0: 0074 6573 7464 6972 5554 0500 01ca 0e23  .testdirUT.....#
000000d0: 6550 4b01 0214 0314 0008 0000 006b a248  ePK..........k.H
000000e0: 57cd b81e ac0e 0000 000e 0000 0010 0009  W...............
000000f0: 0000 0000 0000 0000 00a0 813e 0000 0074  ...........>...t
00000100: 6573 7464 6972 2f74 6573 7466 696c 6555  estdir/testfileU
00000110: 5405 0001 ca0e 2365 504b 0506 0000 0000  T.....#ePK......
00000120: 0200 0200 8500 0000 9300 0000 0000       ..............
```

This occurs for every directory I try this with, on both Gentoo and Ubuntu 22.04.

**Comment by ViRb3** [comment]
Thanks a lot for the report. This is fixed in v1.0.4.

**Comment by fin444** [comment]
Fix works for me, thanks!
